From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: William Douglas <william.douglas@intel.com>
Date: Thu, 14 Jun 2018 21:24:24 +0000
Subject: [PATCH] Move config path to /var/lib/graphviz

The previous config path was in a stateless dir and the config is
needed to be updated after every swupd action potentially so move it
to a statedir instead. Note this will need to be run after creation of
tmpfiles as the config file creation doesn't create directories if
needed.

Also add the tmpfiles configuration for creation the directory.
---
 graphviz.conf      | 1 +
 lib/gvc/gvconfig.c | 4 ++--
 2 files changed, 3 insertions(+), 2 deletions(-)
 create mode 100644 graphviz.conf

diff --git a/graphviz.conf b/graphviz.conf
new file mode 100644
index 0000000..8af9ae0
--- /dev/null
+++ b/graphviz.conf
@@ -0,0 +1 @@
+d /var/lib/graphviz 0755
diff --git a/lib/gvc/gvconfig.c b/lib/gvc/gvconfig.c
index 80d0ce9..5dd916b 100644
--- a/lib/gvc/gvconfig.c
+++ b/lib/gvc/gvconfig.c
@@ -555,8 +555,8 @@ void gvconfig(GVC_t * gvc, boolean rescan)
         }
     
         if (! gvc->config_path) {
-            gvc->config_path = gmalloc(strlen(libdir) + 1 + strlen(config_file_name) + 1);
-            strcpy(gvc->config_path, libdir);
+            gvc->config_path = gmalloc(strlen("/var/lib/graphviz") + 1 + strlen(config_file_name) + 1);
+            strcpy(gvc->config_path, "/var/lib/graphviz");
             strcat(gvc->config_path, DIRSEP);
             strcat(gvc->config_path, config_file_name);
         }
